gnome-initial-setup/pages/keyboard/cc-input-chooser.c

```diff
@@ -621,19 +621,22 @@ fetch_ibus_engines_result (GObject       *object,
                            GAsyncResult  *result,
                            CcInputChooser *chooser)
 {
-        CcInputChooserPrivate *priv = cc_input_chooser_get_instance_private (chooser);
+        CcInputChooserPrivate *priv;
         GList *list, *l;
         GError *error;
 
         error = NULL;
-        list = ibus_bus_list_engines_async_finish (priv->ibus, result, &error);
-        g_clear_object (&priv->ibus_cancellable);
+        list = ibus_bus_list_engines_async_finish (IBUS_BUS (object), result, &error);
         if (!list && error) {
-                g_warning ("Couldn't finish IBus request: %s", error->message);
+                if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
+                        g_warning ("Couldn't finish IBus request: %s", error->message);
                 g_error_free (error);
                 return;
         }
 
+        priv = cc_input_chooser_get_instance_private (chooser);
+        g_clear_object (&priv->ibus_cancellable);
```

Comment:
The last ref on the cancellable is held by the GTask, and it will be freed once you return, correct?